corehq/apps/ota/views.py

```diff
@@ -169,7 +169,14 @@ def phone_holds_all_cases(request):
         except MissingSyncLog:
             return False
 
-    if not case_ids_to_claim and phone_holds_all_cases(request):
+    def cases_to_claim_modified_since_last_synclog_date():
+        try:
+            synclog = get_properly_wrapped_sync_log(request.last_sync_token)
+        except MissingSyncLog:
+            return False
+        return bool(CommCareCase.objects.get_modified_case_ids(domain, list(case_ids_to_claim), synclog))
+
+    if not cases_to_claim_modified_since_last_synclog_date() and phone_holds_all_cases(request):
```

Comment:
I think this should keep the check `not case_ids_to_claim` as well. It might make the code easier to read if we inverted this return logic:
```python
if case_ids_to_claim:
    return HttpResponse(status=201)

if not phone_holds_all_cases(request):
    return HttpResponse(status=201)

if cases_to_claim_modified_since_last_synclog_date():
    return HttpResponse(status=201)

return HttpResponse(status=204)
```

(statements ordered by 'cost' - cheapest first)

Reply:
https://github.com/dimagi/commcare-hq/pull/32554#:~:text=to%20most%20expensive-,1d9e967,-function%20rename -- makes a lot of sense this way